src/prices.go

```diff
@@ -0,0 +1,210 @@
+			}
+		}
+	}
+
+	// Convert to comma-separated list
+	var idList []string
+	for id := range coinIDs {
+		idList = append(idList, id)
+	}
+
+	// Batch fetch all prices
+	url := fmt.Sprintf("https://api.coingecko.com/api/v3/simple/price?ids=%s&vs_currencies=usd",
+		strings.Join(idList, ","))
+
+	debugLog("Fetching all CoinGecko prices", map[string]interface{}{
+		"url":        url,
+		"coin_count": len(idList),
+	})
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return fmt.Errorf("fetching coingecko prices: %v", err)
+	}
+	defer resp.Body.Close()
+
+	var result map[string]map[string]float64
+	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
+		return fmt.Errorf("decoding coingecko response: %v", err)
+	}
+
+	// Cache all prices
+	now := time.Now()
+	for coinID, priceData := range result {
+		if usdPrice, ok := priceData["usd"]; ok {
+			priceCache[coinID] = PriceInfo{
+				Price:     usdPrice,
+				Timestamp: now,
+			}
+		}
+	}
+
+	pricesInitialized = true
+	debugLog("Price cache initialized", map[string]interface{}{
+		"prices_cached": len(priceCache),
+		"timestamp":     now,
+	})
+	return nil
+}
+
+func fetchSkipAssets() error {
+	// Check if cache is still valid
+	if skipCache != nil {
+		if time.Since(skipCache.Timestamp) < PriceCacheTTL {
+			return nil
+		}
+	}
+
+	resp, err := http.Get("https://api.skip.build/v2/fungible/assets")
+	if err != nil {
+		return fmt.Errorf("fetching skip assets: %v", err)
+	}
+	defer resp.Body.Close()
+
+	var skipResp SkipResponse
+	if err := json.NewDecoder(resp.Body).Decode(&skipResp); err != nil {
+		return fmt.Errorf("decoding skip response: %v", err)
+	}
+
+	// Create new cache
+	assets := make(map[string]map[string]SkipAsset)
+	for chainID, chainAssets := range skipResp.ChainToAssetsMap {
+		assets[chainID] = make(map[string]SkipAsset)
+		for _, asset := range chainAssets.Assets {
+			assets[chainID][asset.Denom] = asset
+		}
+	}
+
+	skipCache = &SkipCache{
+		Assets:    assets,
+		Timestamp: time.Now(),
+	}
+
+	// Initialize price cache
+	if err := initializePriceCache(); err != nil {
+		return fmt.Errorf("initializing price cache: %v", err)
+	}
+
+	return nil
+}
+
+func getTokenPrice(coingeckoId string) (float64, error) {
+	debugLog("Getting token price", map[string]string{
+		"token": coingeckoId,
+	})
+
+	// Check cache
+	if info, ok := priceCache[coingeckoId]; ok {
+		if time.Since(info.Timestamp) < PriceCacheTTL {
+			return info.Price, nil
+		}
+	}
+
+	// Cache expired or missing, refresh all prices
+	if err := initializePriceCache(); err != nil {
+		return 0, fmt.Errorf("refreshing price cache: %v", err)
+	}
+
+	// Try cache again after refresh
+	if info, ok := priceCache[coingeckoId]; ok {
+		return info.Price, nil
+	}
+
+	return 0, fmt.Errorf("no price found for token: %s", coingeckoId)
+}
+
+func getAtomPrice() (float64, error) {
+	return getTokenPrice("cosmos")
+}
+
+// Initialize caches in main
+func init() {
+	if err := fetchSkipAssets(); err != nil {
```

Comment:
I think it would be better to call `initializePriceCache()` here instead of `fetchSkipAssets()` and then remove the call to `initializePriceCache()` inside of `fetchSkipAssets()`. WDYT?